extensions/binding/org.eclipse.smarthome.binding.onewire/src/main/java/org/eclipse/smarthome/binding/onewire/internal/handler/EDSSensorThingHandler.java

```diff
@@ -0,0 +1,219 @@
+                        .create(new ChannelUID(getThing().getUID(), CHANNEL_ABSOLUTE_HUMIDITY), "Number:Density")
+                        .withLabel("Abs. Humidity").withType(new ChannelTypeUID(BINDING_ID, "absolutehumidity"))
+                        .build());
+            }
+        } else {
+            if (thing.getChannel(CHANNEL_HUMIDITY) != null) {
+                thingBuilder.withoutChannel(new ChannelUID(getThing().getUID(), CHANNEL_HUMIDITY));
+            }
+            if (thing.getChannel(CHANNEL_DEWPOINT) != null) {
+                thingBuilder.withoutChannel(new ChannelUID(getThing().getUID(), CHANNEL_DEWPOINT));
+            }
+            if (thing.getChannel(CHANNEL_ABSOLUTE_HUMIDITY) != null) {
+                thingBuilder.withoutChannel(new ChannelUID(getThing().getUID(), CHANNEL_ABSOLUTE_HUMIDITY));
+            }
+        }
+
+        // pressure sensor
+        if ((sensorType == OwSensorType.EDS0066) || (sensorType == OwSensorType.EDS0068)) {
+            if (thing.getChannel(CHANNEL_PRESSURE) == null) {
+                thingBuilder.withChannel(
+                        ChannelBuilder.create(new ChannelUID(getThing().getUID(), CHANNEL_PRESSURE), "Number:Pressure")
+                                .withLabel("Pressure").withType(new ChannelTypeUID(BINDING_ID, "pressure")).build());
+            }
+        } else {
+            if (thing.getChannel(CHANNEL_PRESSURE) != null) {
+                thingBuilder.withoutChannel(new ChannelUID(getThing().getUID(), CHANNEL_PRESSURE));
+            }
+        }
+
+        // light sensor
+        if ((sensorType == OwSensorType.EDS0067) || (sensorType == OwSensorType.EDS0068)) {
+            if (thing.getChannel(CHANNEL_LIGHT) == null) {
+                thingBuilder.withChannel(
+                        ChannelBuilder.create(new ChannelUID(getThing().getUID(), CHANNEL_LIGHT), "Number:Illuminance")
+                                .withLabel("Illuminance").withType(new ChannelTypeUID(BINDING_ID, "light")).build());
+            }
+        } else {
+            if (thing.getChannel(CHANNEL_LIGHT) != null) {
+                thingBuilder.withoutChannel(new ChannelUID(getThing().getUID(), CHANNEL_LIGHT));
+            }
+        }
+
+        updateThing(thingBuilder.build());
+
+        validConfig = true;
+        updatePresenceStatus(UnDefType.UNDEF);
+    }
+
+    @Override
+    public Map<String, String> updateSensorProperties(OwBaseBridgeHandler bridgeHandler) throws OwException {
+        Map<String, String> properties = new HashMap<String, String>();
+
+        OwPageBuffer pages = bridgeHandler.readPages(sensorIds.get(0));
+
+        OwSensorType sensorType = OwSensorType.UNKNOWN;
+        try {
+            sensorType = OwSensorType.valueOf(new String(pages.getPage(0), 0, 7, StandardCharsets.US_ASCII));
+        } catch (IllegalArgumentException e) {
+        }
+
+        if ((sensorType != OwSensorType.EDS0064) && (sensorType != OwSensorType.EDS0065)
```

Comment:
Wouldn't it make sense to contains a static collection that contains alls EDS Sensor types and use something like:

```Java
if (!EDSSensorTypes.contains(sensorType)) {
    ...
}
```